pandas/tseries/offsets.py

```diff
@@ -1160,6 +1162,196 @@ def onOffset(self, dt):
     _prefix = 'MS'
 
 
+class SemiMonthOffset(DateOffset):
+    _adjust_dst = True
+    _default_day_of_month = 15
+    _min_day_of_month = 2
+
+    def __init__(self, n=1, normalize=False, **kwds):
+        self.day_of_month = kwds.get('day_of_month',
+                                     self._default_day_of_month)
+
+        if not self._min_day_of_month <= self.day_of_month <= 27:
+            raise ValueError('day_of_month must be '
+                             '{}<=day_of_month<=27, got {}'.format(
+                                 self._min_day_of_month, self.day_of_month))
+        self.n = int(n)
+        self.normalize = normalize
+        self.kwds = kwds
+
+    @classmethod
+    def _from_name(cls, suffix=None):
+        if not suffix:
```

Comment:
oh ok, so continue to have that, just do the init in the constructor as you have now, no need to have same code in 2 places